tests/integration/logging/monolog2/test_monolog_basic.php

```diff
@@ -0,0 +1,198 @@
+  ]
+]
+*/
+
+
+/*EXPECT_LOG_EVENTS
+[
+    {
+      "common": {
+        "attributes": {}
+      },
+      "logs": [
+        {
+          "message": "error",
+          "level": "ERROR",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "critical",
+          "level": "CRITICAL",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "notice",
+          "level": "NOTICE",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "warning",
+          "level": "WARNING",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "info",
+          "level": "INFO",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "alert",
+          "level": "ALERT",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },  
+        {
+          "message": "emergency",
+          "level": "EMERGENCY",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },        
+        {
+          "message": "debug",
+          "level": "DEBUG",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        }
+      ]
+    }
+  ]
+ */
+
+require_once(realpath(dirname(__FILE__)) . '/../../../include/config.php');
+require_once(realpath(dirname(__FILE__)) . '/../../../include/monolog.php');
+require_monolog(2);
+
+use Monolog\Logger;
+use Monolog\Handler\StreamHandler;
+use Monolog\Formatter\LineFormatter;
+
+
+function test_logging() {
```

Comment:
I wonder if it is possible to have this function take a monolog_version parameter, and put it in a reusable module in tests/integration/logging/test_monolog.php. It might reduce some code duplication but I'm not entirely convinced this will be beneficial though.

Reply:
There are strings in the expected metrics which depend on a file path which is different for monolog 2 and monolog 3.  I suppose we could wildcard that?
I would suggest this is a good idea for a future enhancement.